include/net/socket.h

```diff
@@ -688,8 +688,23 @@ static inline ssize_t recv(int sock, void *buf, size_t max_len, int flags)
 	return zsock_recv(sock, buf, max_len, flags);
 }
 
+/*
+ * Need this wrapper because newer GCC versions got too smart and "typecheck"
+ * even macros, so '#define fcntl zsock_fcntl' leads to error.
```

Comment:
That does sound strange, why compiler would check macros in the first place

Reply:
> That does sound strange, why compiler would check macros in the first place


Indeed. You can see yourself how confusing the error message is in the full build log: https://ci.linaro.org/view/lite-iot-ci/job/zephyr-upstream/6180/PLATFORM=frdm_kw41z,ZEPHYR_TOOLCHAIN_VARIANT=zephyr,label=docker-focal-amd64-lite/console . I spent a bunch of time to understand why it happened only for frdm_k41z (turned out it forces newlib via one of its drivers), and then why it happens with only this sample (as https://github.com/zephyrproject-rtos/zephyr/pull/34276 shows, a peculiar include order). Then after playing with various ways to resolve it (nothing worked until I made an aux function with proper prototype), the above is only explanation I can give for gcc behavior. Another turn in compiler vs humans arms race ;-).